src/checker/dd/DDAlternatingChecker.cpp

```diff
@@ -169,4 +147,32 @@ EquivalenceCriterion DDAlternatingChecker::checkEquivalence() {
   return op1.equals(op2);
 }
 
+bool DDAlternatingChecker::canHandle(const qc::QuantumComputation& qc1,
+                                     const qc::QuantumComputation& qc2) {
+  assert(qc1.getNqubits() == qc2.getNqubits());
```

Comment:
Why is this case handled with an assertion? Wouldn't returning `false` achieve the same result?

Reply:
My intention was to sort of ensure that this function is only called in the equivalence checker, when everything is properly set up, i.e., both circuits have been adjusted to have the same number of qubits.
So, if the function were to be called in the right place all the time, that `if` would never be triggered.
In addition, I am a little afraid of the `if` conveying that the alternating checker cannot handle circuits with different number of qubits (which it can; just has to be properly setup).

Does that convince you or would you still rather favor an `if`?

Reply:
So basically that assert shows improper use of the canHandle function. That convinces me :smile: